File: 380-insert-delete-getrandom-o1/insert-delete-getrandom-o1.py
class RandomizedSet:

    def __init__(self):
        self.unique = set()
        
    def insert(self, val: int) -> bool:
        if val in self.unique: return False

        self.unique.add(val)
        return True        

    def remove(self, val: int) -> bool:
        if val not in self.unique: return False

        self.unique.remove(val)
        return True
        

    def getRandom(self) -> int:
        output = list(self.unique)
        randIdx = random.randint(0,len(output)-1) 

        return output[randIdx]

        # SC -> O(N)
        # TC -> O(N)
        


# Your RandomizedSet object will be instantiated and called as such:
# obj = RandomizedSet()
# param_1 = obj.insert(val)
# param_2 = obj.remove(val)
# param_3 = obj.getRandom()

File: 380-insert-delete-getrandom-o1/insert-delete-getrandom-o1.java
class RandomizedSet {
    HashSet<Integer> set;
    Random random;

    public RandomizedSet() {
        set = new HashSet();
        random = new Random();
    }
    
    public boolean insert(int val) {
        if(set.contains(val)) return false;

        set.add(val);

        return true;
        
    }
    
    public boolean remove(int val) {
        if(!set.contains(val)) return false;

        set.remove(val);

        return true;
        
    }
    
    public int getRandom() {
        List<Integer> result = new ArrayList<>(set);

        return result.get(random.nextInt(result.size()));
        
    }
}


// SC -> O(N)
// TC -> O(N)


/**
 * Your RandomizedSet object will be instantiated and called as such:
 * RandomizedSet obj = new RandomizedSet();
 * boolean param_1 = obj.insert(val);
 * boolean param_2 = obj.remove(val);
 * int param_3 = obj.getRandom();
 */